app/cisc/cisc.go

```diff
@@ -0,0 +1,312 @@
+package main
+
+import (
+	"os"
+
+	"encoding/hex"
+
+	"path"
+
+	"io/ioutil"
+	"strings"
+
+	"github.com/dedis/cothority/app/lib/config"
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/services/identity"
+	"gopkg.in/codegangsta/cli.v1"
+)
+
+/*
+Cisc is the Cisc Identity SkipChain to store information in a skipchain and
+being able to retrieve it.
+
+This is only one part of the system - the other part being the cothority that
+holds the skipchain and answers to requests from the cisc-binary.
+*/
+
+var configFile string
+var sshDir string
+var sshConfig string
+var clientApp *identity.Identity
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "SSH keystore client"
+	app.Usage = "Connects to a ssh-keystore-server and updates/changes information"
+	app.Commands = []cli.Command{
+		commandID,
+		commandConfig,
+		commandKeyvalue,
+		commandSSH,
+	}
+	app.Flags = []cli.Flag{
+		cli.IntFlag{
+			Name:  "debug, d",
+			Value: 1,
+			Usage: "debug-level: 1 for terse, 5 for maximal",
+		},
+		cli.StringFlag{
+			Name:  "config, c",
+			Value: "~/.cisc",
+			Usage: "The configuration-directory of cisc",
+		},
+		cli.StringFlag{
+			Name:  "config-ssh, cs",
+			Value: "~/.ssh",
+			Usage: "The configuration-directory of the ssh-directory",
+		},
+	}
+	app.Before = func(c *cli.Context) error {
+		configDir := config.TildeToHome(c.String("config"))
+		os.Mkdir(configDir, 0660)
+		log.SetDebugVisible(c.Int("debug"))
+		configFile = configDir + "/config.bin"
+		if err := loadConfig(); err != nil {
+			log.Error("Problems reading config-file. Most probably you\n",
+				"should start a new one by running with the 'setup'\n",
+				"argument.")
+		}
+		sshDir = config.TildeToHome(c.String("cs"))
+		sshConfig = sshDir + "/config"
+		return nil
+	}
+	app.After = func(c *cli.Context) error {
+		if clientApp != nil {
+			err := saveConfig()
+			log.ErrFatal(err, "Error while creating config-file", configFile)
+		}
+		return nil
+	}
+	app.Run(os.Args)
+
+}
+
+/*
+ * Identity-related commands
+ */
+func idCreate(c *cli.Context) error {
```

Comment:
you never return an error here, but this signature (having an error return) is necessary because of codegangsta. It seems to me that it'd best if you try to follow their **guidance** and return an error instead of panicking. Their library should handle it accordingly (may need some manual test to make sure)